components/link_tooltip/link_tooltip.tsx

```diff
@@ -48,43 +49,49 @@ export default class LinkTooltip extends React.PureComponent<Props, State> {
         };
     }
 
+    componentDidUpdate() {
+        if (this.state.show) {
+            this.tooltipContainer = this.tooltipContainerRef.current;
```

Comment:
Sorry, I meant doing this slightly differently. I meant that we have `showTooltip` just set the state and then `componentDidUpdate` calls `setTimeout` and does the rest of the work.
```javascript
componentDidUpdate(prevProps) {
    if (this.state.show && !prevState.show) { // Also, check that show changed to make sure we don't do this on unrelated updates
        this.setTimeout(() => {
            // Do the other setup stuff here
        }, DELAY);
    }
}

public showTooltip = () => {
    this.setState({show: true});
}
```

That way, we know the ref should always be set by this point, and we're sure to avoid any potential races. We would also need a ref to store the target since we don't have the event any more, but I think that's fine to add as well.

Reply:
Ahh now I see your point. Thanks for further clarification. Changes are done